src/launchbox/token/LaunchboxFactory.sol

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/proxy/Clones.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import {LaunchboxERC20} from "./LaunchboxERC20.sol";
+
+contract LaunchboxFactory is Ownable(msg.sender) {
+    event TokenDeployed(address tokenAddress, address launchboxExchangeAddress);
+    event LaunchboxExchangeImplementationUpdated(address newLaunchboxExchangeAddress);
+    event TokenImplementationUpdated(address newTokenImplementationAddress);
+    event UniswapRouterUpdated(address newRouter);
+
+    address public tokenImplementation;
+    address public launchboxExchangeImplementation;
+    address public uniswapRouter;
+    
+    error EmptyTokenImplementation();
+    error EmptyLaunchboxExchangeImplementation();
+    error EmptyUniswapRouter();
+
+    constructor(address _tokenImplementation, address _launchboxExchangeImplementation, address _uniswapRouter) {
+        if(_tokenImplementation == address(0)) revert EmptyTokenImplementation();
+        if(_launchboxExchangeImplementation == address(0)) revert EmptyLaunchboxExchangeImplementation();
+        if(_uniswapRouter == address(0)) revert EmptyUniswapRouter();
+        tokenImplementation = _tokenImplementation;
+        launchboxExchangeImplementation = _launchboxExchangeImplementation;
+        uniswapRouter = _uniswapRouter;
+    }
+
+    function deployToken(
+        string memory name,
+        string memory symbol,
+        uint256 maxSupply,
+        string memory metadataURI,
+        uint256 marketCapThreshold
```

Comment:
`marketCapThreshold` should not come from the frontend. Ideally, it should be one of the admin parameters to be set in the Exchange contract

Reply:
I thought of keeping it open ended. Also, saw you have written the contracts on another branch. Didn't saw a PR for that so I wrote them too. I'll remove the factory and erc20.

Reply:
Nah, don't remove yours. let's keep yours and just make the changes. Let that open ended decision come from us not the users